File: php/features.php
<?php
declare(strict_types=1);

// InfranodeOpenData SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class InfranodeOpenDataFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new InfranodeOpenDataBaseFeature();
            case "test":
                return new InfranodeOpenDataTestFeature();
            default:
                return new InfranodeOpenDataBaseFeature();
        }
    }
}

File: py/features.py
# InfranodeOpenData SDK feature factory

from feature.base_feature import InfranodeOpenDataBaseFeature
from feature.test_feature import InfranodeOpenDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: InfranodeOpenDataBaseFeature(),
        "test": lambda: InfranodeOpenDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
